fastapi/models/fastapi_endpoint_demo.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2022 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL).
+from enum import Enum
+from typing import List
+
+from odoo import _, api, fields, models
+from odoo.api import Environment
+from odoo.exceptions import AccessError, MissingError, UserError, ValidationError
+
+from odoo.addons.base.models.res_partner import Partner
+
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import APIKeyHeader
+from pydantic import BaseModel, Field
+
+from ..depends import (
+    authenticated_partner,
+    authenticated_partner_from_basic_auth_user,
+    authenticated_partner_impl,
+    fastapi_endpoint,
+    odoo_env,
+)
+
+
+class FastapiEndpoint(models.Model):
+
+    _inherit = "fastapi.endpoint"
+
+    app: str = fields.Selection(
+        selection_add=[("demo", "Demo Endpoint")], ondelete={"demo": "cascade"}
+    )
+    demo_auth_method = fields.Selection(
+        selection=[("api_key", "Api Key"), ("http_basic", "HTTP Bacic")],
+        string="Authenciation method",
+    )
```

Comment:
```suggestion
    )
    registry_sync = fields.Boolean(default=True)
```
If the `registry_sync` field is False, the controller is not registered and all URLs return error 404.

With this change I can test the demo contract/API

Reply:
@flachica The motivation to set this field to false by default is to control when the odoo's routing map is reloaded. The reload of the routing map is costly operation and will impact all the odoo runners. I've to improve the UI to highlight that a sync is required. This filed will disappear from the form in favor of something more user friendly.